### DISCUSSION TITLE
wagmi v2: `writeContract` does not allow passing undefined as an argument?

### DISCUSSION THREAD

**Discussion by POKENA7** [wevm/wagmi#3190] [Q&A] [answered]
I implemented `useSimulateContract` and `useWriteContract`  according to the following document.
https://beta.wagmi.sh/react/guides/migrate-from-v1-to-v2#removed-prepare-hooks

Then, the following warning appeared.
![wagmiv2](https://github.com/wagmi-dev/wagmi/assets/75174441/c2eefc13-7012-4bf6-a94b-e13d4725eb99)

The error can be resolved by checking the `data` is not undefined in advance or by passing `data!!.request`.
These are not mentioned in the documentation, do I need to do so? Or is there any workaround?

**Comment by rawsilence** [comment, marked as answer]
If the button is disabled when `Boolean(data?.request)` is `false` then it's safe to use: `writeContract(data!.request)`

**Comment by POKENA7** [comment]
@danielcrt Thanks for your reply. Surely that is safe! Or maybe we should check `data` isn't `undefined`.
